tests/TestsFoundational.dfy

```diff
@@ -105,4 +106,30 @@ module TestsFoundational {
     var r := new DafnyVMC.DRandomFoundational();
     Tests.TestDiscreteGaussian(1_000_000, r);
   }
+
+  method {:test} TestFisherYatesIdentity()
+    decreases *
+  {
+    var a: array<nat> := new nat[4](i => i); // [0, 1, 2, 3]
+    var r := new DafnyVMC.DRandomFoundational();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
+
+  method {:test} TestFisherYatesConstant()
+    decreases *
+  {
+    var a: array<nat> := new nat[4](i => 0); // [0, 0, 0, 0]
+    var r := new DafnyVMC.DRandomFoundational();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
+
+  method {:test} TestFisherYatesMixed()
+    decreases *
+  {
+    var a: array<nat> := new nat[] [0, 1, 1, 2]; // [0, 1, 1, 2]
+    var r := new DafnyVMC.DRandomFoundational();
+    Tests.TestFisherYates(1_000_000, a, r, Helper.NatToString);
+  }
+
+
```

Comment:
Can you add tests such as array of size 0 or 1, length even or odd, at least one larger array.